browser-sign-in/app/src/main/java/com/okta/android/samples/browser_sign_in/views/UserDashboardFragment.kt

```diff
@@ -0,0 +1,148 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.okta.android.samples.browser_sign_in.views
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.viewModels
+import androidx.navigation.fragment.findNavController
+import com.okta.android.samples.browser_sign_in.R
+import com.okta.android.samples.browser_sign_in.data.LoginState
+import com.okta.android.samples.browser_sign_in.data.UserDashboardViewModel
+import com.okta.android.samples.browser_sign_in.data.UserGreetingState
+import com.okta.android.samples.browser_sign_in.data.UserInfoState
+import com.okta.android.samples.browser_sign_in.databinding.FragmentUserDashboardBinding
+
+class UserDashboardFragment : Fragment() {
+    private val viewModel: UserDashboardViewModel by viewModels()
+    private lateinit var binding: FragmentUserDashboardBinding
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = FragmentUserDashboardBinding.inflate(layoutInflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.logoutButton.setOnClickListener {
+            viewModel.logoutOfBrowser(requireContext())
+        }
+        binding.fetchUserinfoButton.setOnClickListener {
+            viewModel.fetchUserInfo()
+        }
+
+        /**
+         * Perform navigation based on user's login status in the sign-in flow.
+         *
+         * Use an observer to react to updates in the [LoginState]. Updates are asynchronous and are triggered both by user actions,
+         * such as button clicks, and completing the flow.
+         */
+        viewModel.loginState.observe(viewLifecycleOwner) { state ->
+            when (state) {
+                is LoginState.LoggedIn -> {
+                    // Do nothing. UserDashboardFragment is the correct screen for logged in users
+                }
+                is LoginState.LoggedOut -> navigateToLoginScreen()
+            }
+        }
+
+        /**
+         * Update the user interface for changes in the sign-in flow.
+         *
+         * Use an observer to react to updates in [UserGreetingState]. Updates are asynchronous and are triggered both by user actions,
+         * such as button clicks, and completing the flow.
+         */
+        viewModel.userGreetingState.observe(viewLifecycleOwner) { state ->
+            when (state) {
+                is UserGreetingState.Loading -> {
+                    with(binding) {
+                        userGreetingProgressBar.visibility = View.VISIBLE
+                        logoutButton.setBackgroundColor(requireContext().getColor(R.color.gray))
```

Comment:
I think by setting the button `isEnabled = false` it should automatically turn grey.
If it doesn't, we should use a color state list for it instead.

Reply:
Oh cool. I will use color state list for this